open-feature/go-sdk
# feat: Move all files from pkg/openfeature to openfeature.

## This PR


Move the package from being
github.com/open-feature/go-sdk/pkg/openfeature, which has an unnecessary pkg in the import path, to github.com/open-feature/go-sdk/openfeature, without the unnecessary "pkg" in the import path.

The existing github.com/open-feature/go-sdk/pkg/openfeature package is now a compatibility shell; exported types, constants, variables, and functions are now aliased to the new
github.com/open-feature/go-sdk/openfeature equivalents in a way that will pass equality checks, and types are interchangeable between both packages. No logic or tests live in the package anymore, only the exported identifiers, which call through to the new package for behavior.

The memprovider package, similarly, has moved.

### Related Issues


Fixes #227 

### How to test


Any consumers of the package that already exist should be able to transparently use this version (`go mod edit -replace github.com/open-feature/go-sdk=github.com/paddycarver/open-feature@no-pkg`) with no code changes or changes in behavior.
